src/commands/es/get.ts

```diff
@@ -0,0 +1,58 @@
+import { flags } from '@oclif/command'
+import { Kommand } from '../../common'
+import { Client } from '@elastic/elasticsearch'
+
+export default class EsGet extends Kommand {
+  static description = 'Get a document from ES'
+
+  static flags = {
```

Comment:
Since this is a TS file, this should expose an interface for flag named `IEsGetFlag` ?

Reply:
I don't understand because I don't have much experience in TS, I use this `static` definition because it's the way oclif will infer command flags https://oclif.io/docs/flags

Reply:
```javascript
interface LabeledValue {
  label: string;
}

function printLabel(labeledObj: LabeledValue) {
  console.log(labeledObj.label);
}

let myObj = { size: 10, label: "Size 10 Object" };
printLabel(myObj);
```

Interface just serve the purpose of defining the object structure you are manipulating, it does not take any default value nor any description.

```javascript
static flags = {
    help: flags.help(),
    host: flags.string({
        char: 'h',
        description: 'Elasticsearch server host',
        default: process.env.KUZZLE_HOST || 'localhost',
    }),
    port: flags.string({
        char: 'p',
        description: 'Elasticsearch server port',
        default: process.env.KUZZLE_PORT || '9200',
    }),
}
```

will be written in TS (using ocliff type definition)

```ts
import { flags } from '@oclif/command'
import { IOptionFlag, IBooleanFlag } from '@oclif/parser/lib/flags'

export interface IFlags {
    help: IBooleanFlag<void>;
    host: IOptionFlag<string>;
    port: IOptionFlag<string>;
}


export class myclass {
    static description = 'Get a document from ES'
    static flags: IFlags = {
        help: flags.help(),
        host: flags.string({
            char: 'h',
            description: 'Elasticsearch server host',
            default: process.env.KUZZLE_HOST || 'localhost',
        }),
        port: flags.string({
            char: 'p',
            description: 'Elasticsearch server port',
            default: process.env.KUZZLE_PORT || '9200',
        }),
    }
}
```

Reply:
Ok I see but it seems redundant to me to redeclare the types because they already are automatically inferred when the static `flags` variable is declared.  
```
static flags: IFlags = {
        help: flags.help(),
        host: flags.string(...),
  }

flags.help.length // type error from TS because flags.help is a boolean
```

Reply:
ocliff and typescript doesn't serve the same purpose that's why it seem's redundant

- ocliff provide a command line tool helper
- typescript provide the typing that will ensure you are using the proper type and properties for the interface you are using.

Reply:
Yes but my point is oclif methods return strong typescript types